File: Java/RetrofitApps/HeroRetofitApp/app/src/main/java/com/e/vasialeleka/heroretofitapp/Adapter.java
package com.e.vasialeleka.heroretofitapp;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.squareup.picasso.Picasso;

import java.util.List;

public class Adapter extends RecyclerView.Adapter<Adapter.MyViewHolder> {

    List<Hero> list;
    Context context;

    public Adapter(List<Hero> list, Context context) {
        this.list = list;
        this.context = context;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
      View v = LayoutInflater.from(context).inflate(R.layout.item_hero,viewGroup,false);

        return new MyViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder myViewHolder, int i) {
Hero  hero = list.get(i);
myViewHolder.name.setText(hero.getName());
        Picasso.get().load(hero.getImageUrl()).into(myViewHolder.image);

    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public class MyViewHolder extends RecyclerView.ViewHolder {
        TextView name;
        ImageView image;

        public MyViewHolder(@NonNull View itemView) {
            super(itemView);
            name = itemView.findViewById(R.id.name);
            image = itemView.findViewById(R.id.image);
        }


    }

}

File: Kotlin/Rerorofit/app/src/main/java/com/example/vasialeleka/newsappkotlin/Adapter.kt
package com.example.vasialeleka.newsappkotlin

import android.content.Context

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.squareup.picasso.Picasso

class Adapter(internal var list: List<Hero>, internal var context: Context): RecyclerView.Adapter<Adapter.MyViewHolder>() {

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): MyViewHolder {
        val v = LayoutInflater.from(context).inflate(R.layout.item_hero, viewGroup, false)

        return MyViewHolder(v)
    }

    override fun onBindViewHolder(myViewHolder: MyViewHolder, i: Int) {
        val hero = list[i]
        myViewHolder.name.text = hero.name
        Picasso.get().load(hero.imageUrl).into(myViewHolder.image)

    }

    override fun getItemCount(): Int {
        return list.size
    }
    inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        internal var name: TextView
        internal var image: ImageView

        init {
            name = itemView.findViewById(R.id.name)
            image = itemView.findViewById(R.id.image)
        }


    }
}
